python/tests/test_aac/execute/test_plugin_manager.py

```diff
@@ -0,0 +1,25 @@
+from types import ModuleType
+from unittest import TestCase
+
+from aac.execute.plugin_manager import get_plugin_manager, register_plugins_in_package
+
+class TestPluginManager(TestCase):
+    def test_plugin_manager(self):
+        plugin_manager = get_plugin_manager()
+        plugins = plugin_manager.get_plugins()
+        plugin = next(iter(plugins))
+        name = plugin.plugin_name
+        self.assertTrue(str(plugin), ''.join(str(plugins)))
+
+        plugin_manager.unregister(plugin)
+        plugins = plugin_manager.get_plugins()
+        self.assertNotIn(str(plugin), ''.join(str(plugins)))
+
+        plugin_manager.register(plugin)
+        plugins = plugin_manager.get_plugins()
+        self.assertTrue(str(plugin), ''.join(str(plugins)))
+
+    def test_plugin_manager_fail(self):
```

Comment:
So, for some reason when using python -m unittest pluggy will accept non module type arguments.  When using pytest it does not and returns an error.  The pytest functionality seems like it should be the correct one, but this isnt something we can change as it is apart of the pluggy dependency.  Without this test, I really cant think of a way to test a fail case for plugin manager, its a very small file and I dont think users would ever realistically use it directly.  If it can accept anything, then it's pretty much foolproof.

Reply:
Decided to incorporate a comment explaining the current issue implementing more of a failure state test to do investigations at a later time.